src/Cli/Microsoft.TemplateEngine.Cli/Commands/InteractiveMode/InteractiveMode.cs

```diff
@@ -0,0 +1,253 @@
+
+                    IEnumerable<TemplateGroup> templateGroups = TemplateGroup.FromTemplateList(CliTemplateInfo.FromTemplateInfo(templates, hostSpecificDataLoader));
+                    // --- End of env workaround
+
+                    var knownArgs = GetArgs(context.ParseResult);
+                    if (knownArgs is null)
+                    {
+                        return;
+                    }
+
+                    InteractiveQuerying queries = new InteractiveQuerying();
+                    TemplateCommand? templateCommand = GetTemplate(
+                        knownArgs,
+                        settings,
+                        templateGroups,
+                        templatePackageManager);
+
+                    if (templateCommand is null)
+                    {
+                        return;
+                    }
+
+                    context.Console.WriteLine(InteractiveModePrompts.OpeningMessage("<Default Language>", templateCommand.Template.Name, "skip button"));
+
+                    queries.SetQuestions(knownArgs, templateCommand);
+
+                    var questionsToAsk = queries.Questions();
+                    // TODO: adjust to different input types
+                    Dictionary<string, string> paramsToAdd = new Dictionary<string, string>();
+
+                    while (questionsToAsk.MoveNext())
+                    {
+                        var currentQuestion = questionsToAsk.Current;
+                        bool rightInfoCollected = false;
+                        while (!rightInfoCollected)
+                        {
+                            context.Console.WriteLine(currentQuestion.GetQuery());
+                            var paramValue = Console.ReadLine();
+                            if (paramValue is not null
+                                && paramValue != string.Empty
+                                && paramValue.GetType() == currentQuestion.GetValueType())
+                            {
+                                // only supporting string for now
+                                paramsToAdd.Add(currentQuestion.GetParameterName(), paramValue);
+                                rightInfoCollected = true;
+                                //TODO: Necessary checks to see if value by customer is good
+                            }
+                            if (paramValue == string.Empty)
+                            {
+                                rightInfoCollected = true;
+                            }
+                        }
+                    }
+
+                    // Hard coded for now bc I cannot find where this information would be in the parser / parserResult
+                    // Need to get input given by user
+                    string commandLineCall = $"dotnet new {templateCommand.Template.ShortNameList[0]}";
```

Comment:
Consider displaying the final command before running it, so the user can reuse it afterwards.